src/uu/more/src/more.rs

```diff
@@ -105,29 +105,51 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         while let (Some(file), next_file) = (files_iter.next(), files_iter.peek()) {
             let file = Path::new(file);
             if file.is_dir() {
-                terminal::disable_raw_mode().unwrap();
-                return Err(UUsageError::new(
-                    1,
-                    format!("{} is a directory.", file.quote()),
-                ));
+                if length == 1 {
```

Comment:
I'd like to suggest using the `show!` and `map_err_context` functionality:
- https://docs.rs/uucore/latest/uucore/macro.show.html
- https://docs.rs/uucore/latest/uucore/error/trait.FromIo.html

I think those would make it unnecessary to check the number of files and to log the error messages in the buffer.

Reply:
well i was using the buffer so that i dont have to switch the raw mode for each message.. do i even have to do that, since the errormessages are printed to stderr and not stdout? sadly the  documentation for the raw mode doesnt mention stderr.. 
(https://docs.rs/crossterm/0.27.0/crossterm/terminal/index.html#raw-mode)